src/App.tsx

```diff
@@ -24,13 +55,20 @@ function App() {
           `${attributesURL}${dataElementURI}`
         );
         if (response.data[dataElementURI].length === 0) {
-          // TODO: make into a toast
+          addToast({
+            key: uuidv4(),
+            message: `No options found for ${dataElementURI.slice(3)}`,
+            severity: 'info',
+          });
         } else {
```

Comment:
If I recall correctly we tried refactoring that but since the function is asynchronous and returns a promise Typescript/linter doesn't like assigning it directly to a variable and requires an anonymous function that implementation leads to verbosity in addition, if the `fetchOption` function is used inside the useEffect it is required to either be defined inside it as it currently is or be part of the dependency array which would lead to the useEffect running when we don't necessarily want it to.
That being said, I do agree that the function needs to be refactored to handle less responsibility.